lib/grape-swagger.rb

```diff
@@ -45,12 +52,40 @@ def combine_namespaces(app)
                else
                  endpoint.settings.stack.last[:namespace]
                end
-          @combined_namespaces[ns.space] = ns if ns
+          # use the full namespace here (not the latest level only)
+          # and strip leading slash
+          @combined_namespaces[endpoint.namespace.sub(/^\//, '')] = ns if ns
 
           combine_namespaces(endpoint.options[:app]) if endpoint.options[:app]
         end
       end
 
+      def combine_namespace_routes(namespaces)
+        namespaces.each do |name, namespace|
+          parent_route_name = name.match(%r{^/?([^/]*).*$})[1]
+          parent_route = @combined_routes[parent_route_name]
+          namespace_routes = parent_route.collect do |route|
+            route if (route.route_path.start_with?("/#{name}") || route.route_path.start_with?("/:version/#{name}")) &&
+                     (route.instance_variable_get(:@options)[:namespace] == "/#{name}" || route.instance_variable_get(:@options)[:namespace] == "/:version/#{name}")
+          end.compact
+
+          if namespace.options[:nested] == false
```

Comment:
Basically you are right, but this would change the default behavior. If I want to keep the current default behavior I can either check for `if namespace.options[:nested] == false` (current) or `if !namespace.options.key?(:nested) || namespace.options[:nested]` (opposite).

We could also change the key name to sth. like `as_resource: true` or `standalone: true`, but personally I prefer the `nested: false` and would keep the current `if` statement.